src/test/java/io/vlingo/xoom/common/serialization/JsonSerializationTest.java

```diff
@@ -0,0 +1,203 @@
+        executeJsonTest(() -> assertEquals(OffsetDateTime.of(2021, 11, 28, 12, 33, 05, 00, ZoneOffset.of("+02:00")), JsonSerialization.deserialized("\"1638095585000;+02:00\"", (Type) OffsetDateTime.class)));
+    }
+
+    @Test
+    public void canSerializeAList() {
+        List<TestObject> list = new ArrayList<>();
+        list.add(new TestObject("text1", 1));
+        list.add(new TestObject("text2", 2));
+        list.add(new TestObject("text3", 3));
+        executeJsonTest(() -> assertEquals("[{\"myString\":\"text1\",\"myInt\":1},{\"myString\":\"text2\",\"myInt\":2},{\"myString\":\"text3\",\"myInt\":3}]", JsonSerialization.serialized(list)));
+    }
+
+    @Test
+    public void canSerializeACollection() {
+        Collection<TestObject> list = new ArrayList<>();
+        list.add(new TestObject("text1", 1));
+        list.add(new TestObject("text2", 2));
+        list.add(new TestObject("text3", 3));
+        executeJsonTest(() -> assertEquals("[{\"myString\":\"text1\",\"myInt\":1},{\"myString\":\"text2\",\"myInt\":2},{\"myString\":\"text3\",\"myInt\":3}]", JsonSerialization.serialized(list)));
+
+    }
+
+    @Test
+    public void canDeserializeAList() {
+        Collection<TestObject> expectedList = new ArrayList<>();
+        expectedList.add(new TestObject("text1", 1));
+        expectedList.add(new TestObject("text2", 2));
+        expectedList.add(new TestObject("text3", 3));
+
+        executeJsonTest(() -> assertEquals(expectedList, JsonSerialization.deserializedList("[{\"myString\":\"text1\",\"myInt\":1},{\"myString\":\"text2\",\"myInt\":2},{\"myString\":\"text3\",\"myInt\":3}]", new TypeToken<ArrayList<TestObject>>() {
+        }.getType())));
+    }
+
+    @Test
+    @Ignore
+    public void canSerializeCompletes()  {
+        JsonSerialization.setEngine(new GsonJsonSerialization());
+        //assertEquals("{\"state\":{\"futureFactory\":{},\"future\":{\"value\":{\"result\":\"Success\"}},\"failed\":false,\"failureValue\":{},\"handlesFailure\":false,\"id\":{\"id\":\"1\"},\"outcome\":{\"value\":{}},\"outcomeType\":\"Some\",\"timedOut\":false,\"repeats\":false}}", JsonSerialization.serialized(Completes.withSuccess("Success")));
+        JsonSerialization.setEngine(new JacksonJsonSerialization());
+        //assertEquals("{\"state\":{\"futureFactory\":{},\"future\":{\"done\":true,\"numberOfDependents\":0,\"cancelled\":false,\"completedExceptionally\":false},\"failed\":false,\"failureValue\":null,\"handlesFailure\":false,\"id\":{\"id\":\"2\"},\"outcome\":{\"outcome\":\"Success\",\"completed\":true},\"outcomeType\":\"Some\",\"timedOut\":false,\"repeats\":false},\"completed\":true}", JsonSerialization.serialized(Completes.withSuccess("Success")));
+    }
+
+    @Test
+    @Ignore
+    public void canSerializeOptionals() {
+       // assertEquals("{\"myOpt\":\"test\",\"next\":null}", JsonSerialization.serialized(new MyOptional(Optional.of("test"), Optional.empty())));
```

Comment:
Different serialization output

Reply:
How's it different? 

If it's something like spaces between elements, then perhaps we need custom assertions in order to normalise messages before comparing them.

Reply:
One serializer just serializes a given value or null whereas the other serializer uses a sub-structure with a value field for that. Therefore the structures are different and one can't just deserialize the output of one with the other. This results in a compatibility problem. If one starts from scratch this does not seem to be a problem.

Reply:
Makes sense. What's the impact of changed behaviour on end-users?

By the way, are we keeping both serializers? If that's the case it might be a good idea to have a common base test case for both. Differences between implementations could be put into specific test case implementations.

Reply:
Thanks @codekonzept and @jakzal !

I really like the idea of isolating differences in tests.

Reply:
@jakzal: I think it boils down to if we want to be backward compatible or not. If we want to be, leaving it how it is now results in not being able to deserialize existing objects. This is a no go. If we just want to give another serializer option this would be the way to go.

@VaughnVernon + @jakzal: i will work on separating the tests so that the differences in serializing optionals become more obvious.